keyboards/westfoxtrot/cypher/rev1/rev1.c

```diff
@@ -13,7 +13,21 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+#include "rev1.h"
 
-#pragma once
-
-// place overrides here
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        // writePin sets the pin high for 1 and low for 0.
+        // In this example the pins are inverted, setting
+        // it low/0 turns it on, and high/1 turns the LED off.
+        // This behavior depends on whether the LED is between the pin
+        // and VCC or the pin and GND.
+        writePin(F4, !led_state.num_lock);
+        writePin(F1, !led_state.caps_lock);
+        writePin(F5, !led_state.scroll_lock);
+        writePin(B3, !led_state.compose);
+        writePin(B4, !led_state.kana);
```

Comment:
1. Is this the desired behaviour? It seems to be the reverse of what's currently merged into master.
2. B3 and B4 are being used by your switch matrix. Do those LEDs actually exist on the hardware?

Reply:
re 2) this is a mistake on my part from when i copied in the new functions and forgot to remove it.
re 1) I'm sorry I don't know what you mean by the reverse, I'm toggling them incorrectly I assume? so I need to remove the ! from the lines?

Reply:
The `!`s here mean that when the LED state is on (eg. `caps_lock == true`), the pin should be pulled low to turn the LED on (if it's pointing from VCC to the pin). Without them, it would be pulling the pin _high_ to turn the LED on (pointing from GND to the pin instead).

Reply:
What @fauxpark said. The current implementation in `qmk:master` is High-On/Low-Off:

https://github.com/qmk/qmk_firmware/blob/320822d75b785401809f45007320e6fb6885b3fd/keyboards/westfoxtrot/cypher/cypher.c#L39-L56

The change you have here to the new function is **Low**-On/**High**-Off.

Basically, if the Cypher rev1 LEDs toggle correctly on USB LED state (Caps/Num/Scroll Lock) when flashed from current master, the `!`s need to be removed from Lines 26-28 of this branch's `keyboards/westfoxtrot/cypher/rev1/rev1.c`.

Reply:
Ah ok thanks for explaining it for me! sorry for being so slow :-)